### PR TITLE
Cleanup resources properly when Opensearch sink fails to initialize

### PR DESCRIPTION
**Pull request by kkondaka**
### Description
Cleans up resources properly when Opensearch sink fails to initialize
 
### Issues Resolved
Resolves #[Issue number to be closed when this PR is merged]
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ X] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dlvenable** [comment]
@kkondaka , This PR has a lot of commits that are already merged. Can you clean up the branch so that we know what we are reviewing?

**Comment by kkondaka** [comment]
@dlvenable This is strange. I took care of this issue and it showed only two files as modified last time. Not sure how it got changed again

**Review by dlvenable — APPROVED** [review]

File: data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java [hunk 1]
```diff
@@ -615,6 +615,7 @@ private void closeFiles() {
   public void shutdown() {
     super.shutdown();
     closeFiles();
+    openSearchClient.shutdown();
```
Comment (dinujoh) [inline data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java:618]: What does this shutdown api do ? Does it close the client or shutdown the node ?

Comment (kkondaka) [reply data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java:618]: It just shuts down the client to the opensearch. We create it again when we do `doInitialize`. So, shutting down is the correct thing to do.

**Review by dinujoh — APPROVED** [review]

**Pull request merged**